src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -0,0 +1,1065 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Text.Json.Serialization.Metadata;
+using System.Text.Json.Tests;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class DefaultJsonTypeInfoResolverTests
```

Comment:
Adding as comment so we don't forget it. The following test:
```C#
[Fact]
public static void FailingTest()
{
    var resolver = new DefaultJsonTypeInfoResolver { Modifiers = { ContractModifier } };
    var options = new JsonSerializerOptions { TypeInfoResolver = resolver };
    JsonSerializer.Serialize(new PocoWithConstructor("str"), options);    

    static void ContractModifier(JsonTypeInfo jti)
    {
        if (jti.Type == typeof(PocoWithConstructor))
        {
            jti.Properties.Add(jti.CreateJsonPropertyInfo(typeof(string), "someOtherName"));
        }
    }
}

public class PocoWithConstructor
{
    public PocoWithConstructor(string parameter) { }
}
```
Currently fails with 
```
  Message: 
System.ArgumentNullException : Value cannot be null. (Parameter 'obj')

  Stack Trace: 
OrdinalIgnoreCaseComparer.GetHashCode(String obj) line 447
ParameterLookupKey.GetHashCode() line 511
Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior) line 516
Dictionary`2.TryAdd(TKey key, TValue value) line 1120
JsonHelpers.TryAdd[TKey,TValue](Dictionary`2 dictionary, TKey& key, TValue& value) line 24
JsonTypeInfo.InitializeConstructorParameters(JsonParameterInfoValues[] jsonParameters, Boolean sourceGenMode) line 553
JsonTypeInfo.Configure() line 353
ReflectionJsonTypeInfo`1.Configure() line 58
JsonTypeInfo.EnsureConfigured() line 301
JsonSerializerOptions.GetOrAddJsonTypeInfo(Type type) line 44
```
We likely need to expand JsonPropertyInfo test coverage for JsonConstructor types.